tls/s2n_handshake_io.c

```diff
@@ -1032,7 +1032,9 @@ static int s2n_try_delete_session_cache(struct s2n_connection *conn)
 }
 
 int s2n_negotiate(struct s2n_connection *conn, s2n_blocked_status * blocked)
-{    
+{
+    errno = 0;
```

Comment:
Does it make more sense to put this in the `while()` loop? Also, this code has some dependencies on s2n_errno. Can we also set that in the `while()` loop to make sure we don't leak something from the state machine?

Reply:
Ah, there is a macro already `RESET_ERRNO()` while resets the thread locals.